addons/sale/controllers/portal.py

```diff
@@ -73,6 +73,13 @@ def _order_get_page_view_values(self, order, access_token, **kwargs):
 
         return values
 
+    def _get_sortby_orders_values(self):
+        return {
+            'date': {'label': _('Order Date'), 'order': 'date_order desc'},
+            'name': {'label': _('Reference'), 'order': 'name'},
+            'stage': {'label': _('Stage'), 'order': 'state'},
+        }
```

Comment:
For coherence, and to keep overrides working in future versions, you should use `_get_sale_searchbar_sortings` as method name ;).
(See https://github.com/odoo/odoo/commit/c2b0656c12dcba3b564c7a562b78d7ea7092be0d)

Reply:
Done!